src/obstacle_infos_publisher.cpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+#include <yaml-cpp/yaml.h>
+#include <ros/ros.h>
+
+namespace panda_torque_mpc {
+
+class ObstacleInfosPublisher {
+  ObstacleInfosPublisher(
+      const std::shared_ptr<ros::NodeHandle> &ph,
+      const boost::shared_ptr<pinocchio::Model> &pin_model_,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model_);
+  ~ObstacleInfosPublisher();
+  {
+    // Loading the Yaml file
+    YAML::Node config = YAML::LoadFile("config.yaml");
+
+    const YAML::Node& obstacles = config["obstacles"];
+    // Going through all the obstacles and adding them to the geom model
+    for (const auto& obstacle : obstacles)
+    {
+
+    }
+  }
+  void addCollisions();
+  {
+
+  }
+
+  void addObstacle(const std::string &name, const std::string &type,
+                   const Eigen::VectorXd &dim, const Eigen::VectorXd &pose);
```

Comment:
```suggestion
  void addObstacle(const std::string& name, const std::string& type,
                   const Eigen::VectorXd& dim, const Eigen::VectorXd& pose);
```
Same as before with the formatting